PWGJE/Tasks/jetfinderQA.cxx

```diff
@@ -623,19 +624,31 @@ struct JetFinderQATask {
   }
   PROCESS_SWITCH(JetFinderQATask, processJetsMCDWeighted, "jet finder QA mcd with weighted events", false);
 
-  void processJetsMCP(soa::Join<aod::ChargedMCParticleLevelJets, aod::ChargedMCParticleLevelJetConstituents>::iterator const& jet, JetParticles const&)
+  void processJetsMCP(soa::Join<aod::ChargedMCParticleLevelJets, aod::ChargedMCParticleLevelJetConstituents>::iterator const& jet, JetParticles const&, JetMcCollisions const&, soa::Filtered<JetCollisionsMCD> const& collisions)
```

Comment:
put collision at the beggining of the process function. 
I think instead of looping over all collisions for each jet you can slice the collision table based on the jet.McCollisionId(). If there is a McCollision in the slice then you can consider it matched

Reply:
Thanks you for the exact comment.
Possibly, slicing the McCollisions would be needed such like below:
`if (checkMcCollisionsMatched && !jet.mcCollision()**.has_collision()**) {
      return;
    }`
However, as I know, there's no way to call the collision from mcCollision at this moment, right?
I guess I need to declare a new table or something? What do you think of it?
Or if there's a good way to avoid such big task, please let me know.

Reply:
Or another idea I have is slicing the collisions loop such as pseudo code below (Please tell me if you have any idea how to rewrite this with the correct one):
 ` Preslice<JetCollisionsMCD> CollisionsPerMcpJets =  soa::Join<aod::ChargedMCParticleLevelJets, aod::ChargedMCParticleLevelJetConstituents, aod::ChargedMCParticleLevelJetEventWeights>::JMcCollisionId;
  ...
  collisionspermcpejt = collisions.sliceBy(CollisionsPerMcpJets, jet.mcCollision().globalIndex());
  for (auto const& collision : collisions) {
    ...
  }`